provisioning/transport/mqtt/src/mqtt.ts

```diff
@@ -4,12 +4,413 @@
+        /* Codes_SRS_NODE_PROVISIONING_MQTT_18_013: [ When `registrationRequest` receives a successful response from the service, it shall call `callback` passing in null and the response.] */
+        callback(null, result);
+      }
+    };
+
+    /* Codes_SRS_NODE_PROVISIONING_MQTT_18_003: [ `registrationRequest` shall publish to '$dps/registrations/PUT/iotdps-register/?$rid<rid>'.] */
+    this._mqttBase.publish('$dps/registrations/PUT/iotdps-register/?$rid=' + rid, ' ', { qos: 1 } , (err) => {
+      /* Codes_SRS_NODE_PROVISIONING_MQTT_18_004: [ If the publish fails, `registrationRequest` shall wrap the error in a `TransportSpecificError` and call `callback` passing the wrapped error back.] */
+      if (err) {
+        delete this._operations[rid];
+        callback(new errors.TransportSpecificError('publish error', err));
+      }
+    });
+
+    /* Codes_SRS_NODE_PROVISIONING_MQTT_18_007: [ `registrationRequest` shall wait for a response for `timeoutInterval` milliseconds.  After that shall be considered a timeout.] */
+    this._setTimeoutTimer(rid, callback);
+  }
+
+  private _sendOperationStatusQuery(request: RegistrationRequest, operationId: string, callback: (err?: Error, result?: any) => void): void {
+    let rid: string  = uuid.v4();
+
+    /* Codes_SRS_NODE_PROVISIONING_MQTT_18_020: [ `queryOperationStatus` shall wait for a response with a matching rid.] */
+    this._operations[rid] = (err, result) => {
+      delete this._operations[rid];
+      if (err) {
+        /* Codes_SRS_NODE_PROVISIONING_MQTT_18_029: [ When `queryOperationStatus` receives an error from the service, it shall call `callback` passing in the error.] */
+        callback(err);
+      } else {
+        /* Codes_SRS_NODE_PROVISIONING_MQTT_18_027: [ When `queryOperationStatus` receives a successful response from the service, it shall call callback passing in null and the response.] */
+        callback(null, result);
+      }
+    };
+
+    /* Codes_SRS_NODE_PROVISIONING_MQTT_18_017: [ `queryOperationStatus` shall publish to $dps/registrations/GET/iotdps-get-operationstatus/?$rid=<rid>&operationId=<operationId>.] */
+    this._mqttBase.publish('$dps/registrations/GET/iotdps-get-operationstatus/?$rid=' + rid + '&operationId=' + operationId, ' ', { qos: 1 }, (err) => {
+      if (err) {
+        /* Codes_SRS_NODE_PROVISIONING_MQTT_18_018: [ If the publish fails, `queryOperationStatus` shall wrap the error in a `TransportSpecificError` and call callback passing the wrapped error back.] */
+        delete this._operations[rid];
+        callback(new errors.TransportSpecificError('publish error', err));
+      }
+    });
+
+    /* Codes_SRS_NODE_PROVISIONING_MQTT_18_021: [ `queryOperationStatus` shall wait for a response for `timeoutInterval` milliseconds.  After that shall be considered a timeout.] */
+    this._setTimeoutTimer(rid, callback);
+  }
+
+  private _setTimeoutTimer(rid: string, cancelledOperationCallback: (err?: Error, result?: any) => void): void {
```

Comment:
>_setTimeoutTimer [](start = 10, length = 16)

as mentionned in the requirements i would rather not have timeouts in the transport itself and rather handle this in the client level if possible - that way all transports have the same behavior